src/cargo/ops/cargo_compile.rs

```diff
@@ -165,33 +165,62 @@ pub fn compile_pkg<'a>(package: &Package,
         (packages, resolved_with_overrides, registry.move_sources())
     };
 
-    let pkgid = match spec {
-        Some(spec) => try!(resolve_with_overrides.query(spec)),
-        None => package.package_id(),
+    let mut invalid_spec = vec![];
+    let pkgids = if spec.len() > 0 {
+        spec.iter().filter_map(|p| {
+            match resolve_with_overrides.query(&p) {
+                Ok(p) => Some(p),
+                Err(..) => { invalid_spec.push(p.to_string()); None }
+            }
+        }).collect::<Vec<_>>()
+    } else {
+        vec![root_package.package_id()]
     };
-    let to_build = packages.iter().find(|p| p.package_id() == pkgid).unwrap();
-    let targets = try!(generate_targets(to_build, mode, filter, release));
-
-    let target_with_args = match *target_rustc_args {
-        Some(args) if targets.len() == 1 => {
-            let (target, profile) = targets[0];
-            let mut profile = profile.clone();
-            profile.rustc_args = Some(args.to_vec());
-            Some((target, profile))
-        }
-        Some(_) => {
-            return Err(human("extra arguments to `rustc` can only be passed to \
-                              one target, consider filtering\nthe package by \
-                              passing e.g. `--lib` or `--bin NAME` to specify \
-                              a single target"))
+
+    /*
+    if spec.len() > 0 && invalid_spec.len() > 0 {
+        return Err(human(format!("could not find package matching spec `{}`",
+                                 invalid_spec.join(", "))));
+    } */
+
+    let to_builds = packages.iter().filter(|p|
+        pkgids.iter().find(|&op| *op == p.package_id()).is_some()
+    ).collect::<Vec<&Package>>();
+
+    let mut twas = &mut vec![];
```

Comment:
Could this get a more descriptive variable name? Also the `&mut` in front should be fine to omit and  conventionally Cargo uses `Vec::new()` instead of `vec![]`